ops/model.py

```diff
@@ -3598,11 +3598,13 @@ def secret_set(
             args.extend(['--expire', expire.isoformat()])
         if rotate is not None:
             args += ['--rotate', rotate.value]
-        if content is not None:
-            # The content has already been validated with Secret._validate_content
-            for k, v in content.items():
-                args.append(f'{k}={v}')
-        self._run_for_secret('secret-set', *args)
+        with tempfile.TemporaryDirectory() as tmp:
+            # The content is None or has already been validated with Secret._validate_content
+            for k, v in (content or {}).items():
+                with open(f'{tmp}/{k}', mode='w', encoding='utf-8') as f:
```

Comment:
At first I was concerned about slashes in keys, but we (and Juju) is very strict about keys (basically numbers and letters and hyphens) and they've already been checked.

Reply:
Indeed.
I've checked that actually.
The only possible unaccounted for is a key that's very very long, exceeding directory entry name limit, as we don't limit the length in ops.
I figure it's OK to fail in that case.